crates/cast/src/script.rs

```diff
@@ -0,0 +1,360 @@
+        OrderedHashMap::default(),
+    )
+    .with_context(|| "Failed setting up runner.")?;
+
+    let func = runner.find_function("::main")?;
+    let (entry_code, builtins) = runner.create_entry_code(func, &Vec::new(), usize::MAX)?;
+    let footer = runner.create_code_footer();
+    let instructions = chain!(
+        entry_code.iter(),
+        runner.get_casm_program().instructions.iter(),
+        footer.iter()
+    );
+
+    // import from cairo-lang-runner
+    let (hints_dict, string_to_hint) = build_hints_dict(instructions.clone());
+
+    // hint processor
+    let block_context = build_block_context(CheatnetBlockInfo::default());
+    let account_context = build_transaction_context();
+    let mut context = EntryPointExecutionContext::new(
+        &block_context.clone(),
+        &account_context,
+        ExecutionMode::Execute,
+    );
+
+    let mut blockifier_state =
+        CachedState::new(DictStateReader::default(), GlobalContractCache::default());
+    let mut execution_resources = ExecutionResources::default();
+
+    let syscall_handler = SyscallHintProcessor::new(
+        &mut blockifier_state,
+        &mut execution_resources,
+        &mut context,
+        // This segment is created by SierraCasmRunner
+        Relocatable {
+            segment_index: 10,
+            offset: 0,
+        },
+        CallEntryPoint::default(),
+        &string_to_hint,
+        ReadOnlySegments::default(),
+    );
+
+    let mut cairo_hint_processor = CairoHintProcessor {
+        blockifier_syscall_handler: syscall_handler,
+        hints: &string_to_hint,
+        provider,
+        runtime,
+    };
+
+    match runner.run_function(
+        runner.find_function("::main")?,
+        &mut cairo_hint_processor,
+        hints_dict,
+        instructions,
+        builtins,
+    ) {
+        Ok(result) => match result.value {
+            RunResultValue::Success(data) => Ok(ScriptResponse {
+                status: "success".to_string(),
+                msg: build_readable_text(&data),
+            }),
+            RunResultValue::Panic(panic_data) => Ok(ScriptResponse {
+                status: "script panicked".to_string(),
+                msg: build_readable_text(&panic_data),
+            }),
+        },
+        Err(err) => Err(err.into()),
+    }
+}
+
+// taken from starknet-foundry/crates/forge/src/test_case_summary.rs
+/// Helper function to build `readable_text` from a run data.
+fn build_readable_text(data: &Vec<Felt252>) -> Option<String> {
+    let mut readable_text = String::new();
+
+    for felt in data {
+        readable_text.push_str(&format!("\n    original value: [{felt}]"));
+        if let Some(short_string) = as_cairo_short_string(felt) {
+            readable_text.push_str(&format!(", converted to a string: [{short_string}]"));
+        }
+    }
+
+    if readable_text.is_empty() {
+        None
+    } else {
+        readable_text.push('\n');
+        Some(readable_text)
+    }
+}
```

Comment:
Instead of duplicating it, we could consider moving it to a separate common crate in another PR.

Reply:
added an issue https://github.com/foundry-rs/starknet-foundry/issues/1127